org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/StandardPreferenceManager.java

```diff
@@ -102,24 +107,52 @@ public void update(Preferences preferences) {
 		try {
 			Settings mavenSettings = MavenPlugin.getMaven().getSettings();
 			boolean oldDisableTest = false;
+			String multiModuleProjectDirectory = null;
+			String systemMmpd = System.getProperty(MAVEN_MULTI_MODULE_PROJECT_DIRECTORY);
 			List<String> activeProfilesIds = mavenSettings.getActiveProfiles();
 			for (org.apache.maven.settings.Profile settingsProfile : mavenSettings.getProfiles()) {
 				if ((settingsProfile.getActivation() != null && settingsProfile.getActivation().isActiveByDefault()) || activeProfilesIds.contains(settingsProfile.getId())) {
 					if (TRUE.equals(settingsProfile.getProperties().get(M2E_DISABLE_TEST_CLASSPATH_FLAG))) {
 						oldDisableTest = true;
+					}
+					if (systemMmpd != null) {
+						Object mmpd = settingsProfile.getProperties().get(MAVEN_MULTI_MODULE_PROJECT_DIRECTORY);
+						if (mmpd instanceof String s) {
+							multiModuleProjectDirectory = s;
+						}
+					}
+					if (oldDisableTest && multiModuleProjectDirectory != null) {
 						break;
 					}
 				}
 			}
-			if (oldDisableTest != preferences.isMavenDisableTestClasspathFlag()) {
+			if ((systemMmpd == null && multiModuleProjectDirectory == null) || (oldDisableTest != preferences.isMavenDisableTestClasspathFlag())) {
 				mavenSettings.getProfiles().removeIf(p -> JAVALS_PROFILE.equals(p.getId()));
-				if (preferences.isMavenDisableTestClasspathFlag()) {
+				if (preferences.isMavenDisableTestClasspathFlag() || ((systemMmpd == null && multiModuleProjectDirectory == null))) {
 					Profile profile = new Profile();
 					profile.setId(JAVALS_PROFILE);
 					Activation activation = new Activation();
 					activation.setActiveByDefault(true);
 					profile.setActivation(activation);
 					profile.getProperties().put(M2E_DISABLE_TEST_CLASSPATH_FLAG, String.valueOf(preferences.isMavenDisableTestClasspathFlag()));
+					if (preferences.getRootPaths() != null) {
+						for (IPath path : preferences.getRootPaths()) {
+							File f = MavenProperties.computeMultiModuleProjectDirectory(path.toFile());
```

Comment:
In a simple Maven project, I noticed this just returned `null`. Is this expected ?